O365.Security.Native.ETW/UserTrace.hpp

```diff
@@ -122,11 +122,22 @@ namespace Microsoft { namespace O365 { namespace Security { namespace ETW {
 
     inline UserTrace::~UserTrace()
     {
-        if (disposed_) {
+        if (disposed_)
+        {
             return;
         }
 
-        Stop();
+        try
+        {
+            Stop();
+        }
+        catch (...)
+        {
+            // Stop may throw if the trace has been removed by another process.
+            // We catch and swallow the exception here to avoid an exception from leaking
```

Comment:
A blanket `catch (...)` makes me nervous. Given we know what the exception type could be that is problematic, it's probably better to catch those specific exceptions and let the remainder surface. Otherwise, you may be hiding/obscuring a more serious problem.

For example:

```
try {
    Stop();
}
catch (const krabs::no_trace_sessions_remaining&) {
    // comment about not throwing in this scenario
}
```